apps/ledger-live-mobile/src/screens/Portfolio/index.tsx

```diff
@@ -58,9 +65,41 @@ function PortfolioScreen({ navigation }: NavigationProps) {
   const { t } = useTranslation();
 
   const discreetMode = useSelector(discreetModeSelector);
+  const hasBeenUpsoldProtect = useSelector(hasBeenUpsoldProtectSelector);
+  const lastConnectedDevice = useSelector(lastConnectedDeviceSelector);
   const [isAddModalOpened, setAddModalOpened] = useState(false);
   const { colors } = useTheme();
   const { isAWalletCardDisplayed } = useDynamicContent();
+  const protectFeature = useFeature("protectServicesMobile");
+  const postOnboardingURL =
+    protectFeature?.params?.onboardingRestore?.postOnboardingURI;
+  const dispatch = useDispatch();
+
+  const internetReachable = async () => {
+    // workaround : on iOS, netInfo returns null on the first call, see ; https://github.com/react-native-netinfo/react-native-netinfo/issues/572
+    let isInternetReachable = false;
+
+    isInternetReachable = !!(await NetInfo.fetch()).isInternetReachable;
+
+    if (isInternetReachable) {
+      return isInternetReachable;
+    }
+    isInternetReachable = !!(await NetInfo.fetch()).isInternetReachable;
+    return isInternetReachable;
+  };
+
+  useEffect(() => {
+    const openProtectUpsell = async () => {
+      const internetConnected = await internetReachable();
+      if (internetConnected && postOnboardingURL) {
+        Linking.openURL(postOnboardingURL);
+      }
+    };
+    if (!hasBeenUpsoldProtect && lastConnectedDevice?.modelId === "nanoX") {
+      openProtectUpsell();
+      dispatch(setHasBeenUpsoldProtect(true));
```

Comment:
in this case you still dispatch even if there's no internet and if there's no `postOnboardingURL`, maybe move that dispatch to `openProtectUpsell`

Reply:
I'm not sure about that, it may cause having the upsell popping randomly long after the onboarding (when you get internet back). I made this to ensure there's only one point where we control when it displays.

Reply:
👍 makes sense, then it has only one opportunity to be displayed. But that goes along with the spec of "displaying it at the end of the onboarding".